packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -127,6 +127,13 @@ export interface StateMachineProps {
    * @default false
    */
   readonly tracingEnabled?: boolean;
+
+  /**
+   * The removal policy to apply to state machine
+   *
+   * @default RemovalPolicy.DESTROY
+   */
+  readonly removalPolicy?: RemovalPolicy;
```

Comment:
Would adding this props trigger a replacement on the resource? If so, adding a removal policy to retain would, actually delete what they're trying to retain.

Reply:
No, it doesn't. It's a CF attribute.